### ISSUE TITLE
GA release for April 2013

### ISSUE THREAD

**Issue by sskhiri** [roq-messaging/RoQ#119]
# Objective

We have been incubating RoQ for a little bit more than 1 year, it is time now to have our first 1.0 GA version. 
This version will be just a subset of the core stable features.
# Features of the GA
- Static event message queue: a user can manually add host and exchanges and the system will use them to balance the load, but this will not be automatic in the 1.0 GA?
- No management console, just the script released for testing and perhaps the CLI developed by Jehan (TBD).
# Actions to be taken
1. Testing static behavior as described in task #117 @sskhiri
2. Adding a job on the CI for the master branch (not need to have a nightly build just a manual one) @jbruggem 
3. Updating the GimmeRoQ.sh for integrating the Release build from the CI (2)
4. Re-Write a simplified version of the Readme.MD and adding the documentation on the CLI that can be found on the back-end @sskhiri
5. Packaging the CLI with the backend in nice way @jbruggem 
6. Find an agreement on the release process: tag? branch?

Finally, if we can go forward with the management console it would be great. We need to take technology decision asap on that point.

**Comment by jbruggem** [comment]
Additional note: Rather than removing what's in README.md, we should move README.md to INSTALL.md, write a simpler README.md and reference INSTALL.md in README.md (as "manual install" link).

**Comment by cduverne** [comment]
Release name proposal : Marina !!!!!!

**Comment by sskhiri** [comment]
@jbruggem as soon as the issue #122 is finished @cduverne will be able to update and test its issue #121. Thank you !!

**Comment by sskhiri** [comment]
@jbruggem do you want to include the roq-backend in the first release ?

**Comment by jbruggem** [comment]
Yes, I think it's worth it. The CLI works pretty well and the REST API is clean.

**Comment by sskhiri** [comment]
OK good! Can you check if you it is well updated with the ZMQ 3.X and that the GimmeRoQ.sh will well install it. @cduverne were mentioning issues whith NPM. Can you check and confirm ?

Then, the only thing I will do is writing a short wiki page about launching and usage of the backend CLI. Therefore, I can officially close the ticket.

**Comment by sskhiri** [comment]
OK good! Can you check if you it is well updated with the ZMQ 3.X and that the GimmeRoQ.sh will well install it. @cduverne were mentioning issues whith NPM. Can you check and confirm ?

Then, the only thing I will do is writing a short wiki page about launching and usage of the backend CLI. Therefore, I can officially close the ticket.

**Comment by Majea** [comment]
This week I've put my hands on backbone.js. It's pretty nice for organizing
your code but it won't help for having a UI that looks nice. Like all
frameworks I've seen until now, it doesn't support complex models with
multiple references to a single object. But I don't think it's a common
use-case for management UIs. So I guess we don't need that feature here and
we can adopt backbone for MVC implementation.
Still, other UI libs have to be investigated to improve the look of the
console.

Max

**Comment by jbruggem** [comment]
@Majea, you're a bit out of topic there :). 

@sskhiri indeed, the backend should be updated to fix dependency problems. it's in my TODO. 

@sskhiri may I remind you that the doc exists, now :)
See: 
[main page](https://github.com/roq-messaging/roq-backend/blob/master/README.md)
[CLI manual](https://github.com/roq-messaging/roq-backend/blob/master/docs/MANUAL.cli.md)
[API manual](https://github.com/roq-messaging/roq-backend/blob/master/docs/MANUAL.api.md)
[Internal architecture](https://github.com/roq-messaging/roq-backend/blob/master/docs/architecture.md)

**Comment by sskhiri** [comment]
# Status of this task

## DONE
1. Testing static behavior as described in task #117. This has shown few important bugs that made the exchange static wrong and the re-location fail. Now that's working fine!
2. Adding a job on the CI for the master branch 
3. Updating the GimmeRoQ.sh for integrating the Release build from the CI (2)
4. Re-Write a simplified version of the Readme.MD and adding the documentation on the CLI that can be found on the back-end 

## Still TODO or need to be closed
1. The tagging policy on developer handbook
2. Issues #121 #124 #125 
3. Update of the Backend and CLI + Validation for ZMQ 3.0 @jbruggem 
4. The new management console in JAVA @Majea

**Comment by sskhiri** [comment]
@jbruggem still working on #125.

**Comment by sskhiri** [comment]
# New GA objective

Date: March 1 2014
Content:
We need to provide:
- The new chef recipe installation with its guide @jbruggem 
- The upgrade of the CLI with ZMQ 3.X @jbruggem 
- Global validation of the installation @sskhiri
